### ISSUE TITLE
[0.2.1-deb] Crasher: Download after deletion

### ISSUE THREAD

**Issue by eloquence** [freedomofpress/securedrop-client#857]
Encountered [during QA](https://github.com/freedomofpress/securedrop-workstation/issues/484).

1. A deletion job failed on the first try.
2. It eventually succeeded, but hit an AppArmor issue (#856).
3. A subsequent download attempt of a file on a _different_ source caused a crash , see [logs](https://gist.github.com/eloquence/9d8ad48572a102c676f712123841af2b). (For the avoidance of confusion: the test file was in fact an `.so` file.)

**Comment by redshiftzero** [comment]
hmm this seems related to #717

**Comment by redshiftzero** [comment]
Tried a couple things:
 * I thought this might be very similar to #717, so I deleted a source while a long-running file download was in progress. This worked as expected, the source is deleted, the file download fails but the user is notified gracefully, the job is removed from the queue and no crash occurs.
* Next I thought perhaps a `FileWidget` from the deleted source was not getting deleted for some reason, and so the `on_file_download` slot was running, referencing a `File` object that no longer exists and causing the crasher (this does seem from the traceback to be what is happening). Tried this scenario in both my dev env and using the `python3-pyqt5` package (in case there was divergence from `pyqt5`), but all works as expected. No errors or crashes. 

Regardless of this specific bug it is problematic that an exception occurring in a slot will apparently crash the entire app, we should fix that to log any exceptions.
